File: scripts/add-console-cover-metadata.py
import json
import pathlib
import re

path = pathlib.Path('data') / 'games.json'
text = path.read_text(encoding='utf-8')
data = json.loads(text)
console_platforms = {'PS1', 'PS2', 'PS3', 'PSP', 'VITA', 'NDS', 'N3DS', 'WII', 'WIIU', 'NSW'}


def slug(s):
    s = str(s or '')
    s = re.sub(r"[’'\"?!.:,/&]", '', s)
    s = re.sub(r"\s+", '-', s)
    s = re.sub(r"[^a-z0-9\-]", '', s.lower())
    s = re.sub(r"-+", '-', s)
    return s.strip('-')

updated = 0
for game in data.get('GAMES', []):
    if not isinstance(game, dict):
        continue
    p = str(game.get('p', 'PC') or 'PC').upper()
    if p in console_platforms and not game.get('cover') and not game.get('consoleCover'):
        game['consoleCover'] = f"covers/{p.lower()}/{slug(game.get('t'))}.jpg"
        updated += 1

path.write_text(json.dumps(data, indent=2, ensure_ascii=False) + '\n', encoding='utf-8')
print(f'Updated {updated} console games with consoleCover metadata')

File: scripts/add-console-cover-metadata.js
const fs = require('fs');
const path = require('path');
const file = path.join(__dirname, '..', 'data', 'games.json');
const text = fs.readFileSync(file, 'utf8');
const data = JSON.parse(text);
const consolePlatforms = new Set(['PS1', 'PS2', 'PS3', 'PSP', 'VITA', 'NDS', 'N3DS', 'WII', 'WIIU', 'NSW']);
function slug(s) {
  return String(s || '')
    .toLowerCase()
    .replace(/[’'"?!.:,\/&]/g, '')
    .replace(/\s+/g, '-')
    .replace(/[^a-z0-9\-]/g, '')
    .replace(/-+/g, '-')
    .replace(/^-|-$/g, '');
}
let updated = 0;
for (const game of data.GAMES || []) {
  if (!game || typeof game !== 'object') continue;
  const p = String(game.p || 'PC').toUpperCase();
  if (consolePlatforms.has(p) && !game.cover && !game.consoleCover) {
    game.consoleCover = `covers/${p.toLowerCase()}/${slug(game.t)}.jpg`;
    updated += 1;
  }
}
fs.writeFileSync(file, JSON.stringify(data, null, 2) + '\n', 'utf8');
console.log(`Updated ${updated} console games with consoleCover metadata`);
